internal/pkg/deploy/cloudformation/stack/app.go

```diff
@@ -33,17 +33,51 @@ type AppResourcesConfig struct {
 	Version  int                   `yaml:"Version"`
 }
 
-// AppResourcesService is a configuration for service for a deployed Application StackSet
+// AppResourcesService is a union type of either legacy or new
+// service configuration for a deployed Application StackSet
 type AppResourcesService struct {
+	Name   *LegacyAppResourcesServiceConfig `yaml:"Name,flow"`
+	Config AppResourcesServiceConfig
+}
+
+// LegacyAppResourcesServiceConfig is the legacy service configuration for a deployed Application StackSet
+type LegacyAppResourcesServiceConfig string
+
+// AppResourcesServiceConfig is a service configuration for a deployed Application StackSet
+type AppResourcesServiceConfig struct {
```

Comment:
nit: lowercase the yaml tags (`"name", "withECR"`)

Reply:
We might want to make them consistent? It's capitalizing all the other fields

Reply:
ah ok, makes sense!